cmd/pomerium-cli/cache_test.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// creating Mock Cache operations
+type MockCache struct {
+	mock.Mock
+}
+
+// mock func for cache.ExecCredentialsPath()
+func (m *MockCache) ExecCredentialsPath() (string, error) {
+	args := m.Called()
+	return args.String(0), args.Error(1)
+}
+
+func TestClearAllCachedCredentials(t *testing.T) {
+	filePath := t.TempDir()
+	//mock object
+	mockCache := new(MockCache)
+
+	//set up dir expectation for ExecCredentialsPath
+	mockCache.On(" ExecCredentialsPath").Return(filePath+"/tmp/cache", nil)
+	_, err := os.Create(filePath + "/tmp/cache/file1.json")
+	if err != nil {
+		t.Log(err)
+		return
+	}
```

Comment:
By ignoring this error, the test can pass without ever exercising the behavior of `clearAllCachedCredentials()`. In general it would probably be better to do something like `require.NoError(t, err)` instead (adding an import for `"github.com/stretchr/testify/assert"`).

When running this test I found that `file1.json` could not be created because the `tmp` and `cache` directories did not exist.

Reply:
i decided to use a `tempdir` so we wouldnt have unwanted test files generated. 
If you want  the files to be created i can implement that